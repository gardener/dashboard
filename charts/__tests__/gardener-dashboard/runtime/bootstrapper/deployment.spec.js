//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { helm } = fixtures

const renderTemplates = helm.renderBootstrapperRuntimeTemplates

describe('gardener-dashboard-terminal-bootstrapper', () => {
  describe('deployment', () => {
    let values
    let templates

    beforeEach(() => {
      values = {
        global: {
          bootstrapper: {
            enabled: true
          }
        }
      }
      templates = [
        'deployment'
      ]
    })

    it('should not render the template', async function () {
      values.global.bootstrapper.enabled = false
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [deployment] = documents
      expect(deployment).toBe(null)
    })

    it('should render the template with default values', async () => {
      values.global.bootstrapper.kubeconfig = 'apiVersion: v1'
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [deployment] = documents
      expect(deployment).toMatchSnapshot({
        spec: {
          template: {
            metadata: {
              annotations: {
                'checksum/configmap-gardener-dashboard-terminal-bootstrapper-config': expect.stringMatching(/[0-9a-f]{64}/),
                'checksum/secret-gardener-dashboard-terminal-bootstrapper-kubeconfig': expect.stringMatching(/[0-9a-f]{64}/)
              }
            }
          }
        }
      })
    })

    it('should render the template with pod labels and annotations', async () => {
      const labels = {
        b: 'foo'
      }
      const annotations = {
        a: 'bar'
      }
      values.global.bootstrapper.podLabels = labels
      values.global.bootstrapper.podAnnotations = annotations
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [deployment] = documents
      expect(deployment.spec.template.metadata.labels).toEqual(expect.objectContaining(labels))
      expect(deployment.spec.template.metadata.annotations).toEqual(expect.objectContaining(annotations))
    })
  })
})
