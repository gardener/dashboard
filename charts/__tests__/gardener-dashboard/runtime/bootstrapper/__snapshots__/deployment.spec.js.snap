// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard-terminal-bootstrapper deployment should render the template with default values 1`] = `
Object {
  "apiVersion": "apps/v1",
  "kind": "Deployment",
  "metadata": Object {
    "labels": Object {
      "app.kubernetes.io/component": "terminal-bootstrapper",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-runtime-0.1.0",
    },
    "name": "gardener-dashboard-terminal-bootstrapper",
    "namespace": "garden",
  },
  "spec": Object {
    "replicas": 1,
    "selector": Object {
      "matchLabels": Object {
        "app.kubernetes.io/component": "terminal-bootstrapper",
        "app.kubernetes.io/name": "gardener-dashboard",
      },
    },
    "template": Object {
      "metadata": Object {
        "annotations": Object {
          "checksum/configmap-gardener-dashboard-terminal-bootstrapper-config": StringMatching /\\[0-9a-f\\]\\{64\\}/,
          "checksum/secret-gardener-dashboard-terminal-bootstrapper-kubeconfig": StringMatching /\\[0-9a-f\\]\\{64\\}/,
        },
        "labels": Object {
          "app.kubernetes.io/component": "terminal-bootstrapper",
          "app.kubernetes.io/instance": "gardener-dashboard",
          "app.kubernetes.io/managed-by": "Helm",
          "app.kubernetes.io/name": "gardener-dashboard",
          "helm.sh/chart": "gardener-dashboard-runtime-0.1.0",
        },
      },
      "spec": Object {
        "containers": Array [
          Object {
            "args": Array [
              "--max-old-space-size=460",
              "main.js",
            ],
            "env": Array [
              Object {
                "name": "GARDENER_CONFIG",
                "value": "/etc/gardener-dashboard-terminal-bootstrapper/config/config.yaml",
              },
              Object {
                "name": "KUBECONFIG",
                "value": "/etc/gardener-dashboard-terminal-bootstrapper/kubeconfig/kubeconfig.yaml",
              },
            ],
            "image": "eu.gcr.io/gardener-project/gardener/dashboard-terminal-bootstrapper:latest",
            "imagePullPolicy": "IfNotPresent",
            "livenessProbe": Object {
              "failureThreshold": 3,
              "httpGet": Object {
                "path": "/live",
                "port": 9000,
              },
              "initialDelaySeconds": 10,
              "periodSeconds": 30,
              "successThreshold": 1,
              "timeoutSeconds": 5,
            },
            "name": "gardener-dashboard-terminal-bootstrapper",
            "ports": Array [
              Object {
                "containerPort": 9000,
                "name": "http",
                "protocol": "TCP",
              },
            ],
            "readinessProbe": Object {
              "failureThreshold": 1,
              "httpGet": Object {
                "path": "/ready",
                "port": 9000,
              },
              "initialDelaySeconds": 5,
              "periodSeconds": 5,
              "successThreshold": 1,
              "timeoutSeconds": 5,
            },
            "resources": Object {
              "requests": Object {
                "cpu": "100m",
                "memory": "100Mi",
              },
            },
            "volumeMounts": Array [
              Object {
                "mountPath": "/etc/gardener-dashboard-terminal-bootstrapper/config",
                "name": "gardener-dashboard-terminal-bootstrapper-config",
                "readOnly": true,
              },
              Object {
                "mountPath": "/etc/gardener-dashboard-terminal-bootstrapper/kubeconfig",
                "name": "gardener-dashboard-terminal-bootstrapper-kubeconfig",
                "readOnly": true,
              },
              Object {
                "mountPath": "/var/run/secrets/projected/serviceaccount",
                "name": "service-account-token",
                "readOnly": true,
              },
            ],
          },
        ],
        "restartPolicy": "Always",
        "serviceAccountName": "gardener-dashboard-terminal-bootstrapper",
        "volumes": Array [
          Object {
            "configMap": Object {
              "name": "gardener-dashboard-terminal-bootstrapper-configmap-config",
            },
            "name": "gardener-dashboard-terminal-bootstrapper-config",
          },
          Object {
            "name": "gardener-dashboard-terminal-bootstrapper-kubeconfig",
            "secret": Object {
              "secretName": "gardener-dashboard-terminal-bootstrapper-secret-kubeconfig",
            },
          },
          Object {
            "name": "service-account-token",
            "projected": Object {
              "sources": Array [
                Object {
                  "serviceAccountToken": Object {
                    "expirationSeconds": 43200,
                    "path": "token",
                  },
                },
              ],
            },
          },
        ],
      },
    },
  },
}
`;
